src/modeler/AutoRest.Swagger/Validation/TrackedResourceGetOperationValidation.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model.Utilities;
+using System.Collections.Generic;
+using AutoRest.Swagger.Model;
+using System.Text.RegularExpressions;
+using System.Linq;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class TrackedResourceGetOperationValidation : TypedRule<Dictionary<string, Schema>>
+    {
+        private readonly Regex resNames = new Regex(@"(RESOURCE|TRACKEDRESOURCE)$", RegexOptions.IgnoreCase);
+
+        /// <summary>
+        /// The template message for this Rule. 
+        /// </summary>
+        /// <remarks>
+        /// This may contain placeholders '{0}' for parameterized messages.
+        /// </remarks>
+        public override string MessageTemplate => "Tracked resource {0} must have a get operation.";
+
+        /// <summary>
+        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// </summary>
+        public override Category Severity => Category.Warning;
+
+        // Verifies if a tracked resource has a corresponding get operation
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> definitions, RuleContext context)
```

Comment:
This is similar to a number of validation rules already implemented. Please refer [https://github.com/Azure/autorest/blob/22f6f44aa8ca6560b581f8748c5fed958a73aa83/src/modeler/AutoRest.Swagger/Validation/ModelTypeIncomplete.cs#L19](https://github.com/Azure/autorest/blob/22f6f44aa8ca6560b581f8748c5fed958a73aa83/src/modeler/AutoRest.Swagger/Validation/ModelTypeIncomplete.cs#L19) This does return a validation error message per violation.

---
In reply to: [102595319](https://github.com/Azure/autorest/pull/1865#discussion_r102595319) [](ancestors = 102595319)